pennylane/transforms/defer_measurements.py

```diff
@@ -182,6 +190,10 @@ def func(x, y):
         if isinstance(op, MidMeasureMP):
             _ = measured_wires.pop(0)
 
+            if op.postselect is not None:
+                with QueuingManager.stop_recording():
+                    new_operations.append(qml.Projector([float(op.postselect)], wires=op.wires[0]))
```

Comment:
why do you need to cast to float?

Reply:
I didn't need to, but matrix dtypes have caused trouble in the past so I figured why not. I can remove it and if everything works then great :)

Reply:
Removed.